src/ImageProcessor/Imaging/ResizeLayer.cs

```diff
@@ -102,67 +103,49 @@ public class ResizeLayer
-                && this.Upscale == resizeLayer.Upscale
-                && ((this.CenterCoordinates != null
-                    && resizeLayer.CenterCoordinates != null
-                    && this.CenterCoordinates.SequenceEqual(resizeLayer.CenterCoordinates))
-                    || (this.CenterCoordinates == resizeLayer.CenterCoordinates))
-                && this.MaxSize == resizeLayer.MaxSize
-                && ((this.RestrictedSizes != null
-                    && resizeLayer.RestrictedSizes != null
-                    && this.RestrictedSizes.SequenceEqual(resizeLayer.RestrictedSizes))
-                    || (this.RestrictedSizes == resizeLayer.RestrictedSizes))
-                && this.AnchorPoint == resizeLayer.AnchorPoint;
-        }
-
-        /// <summary>
-        /// Returns the hash code for this instance.
-        /// </summary>
+        /// <summary>Indicates whether the current object is equal to another object of the same type.</summary>
+        /// <param name="other">An object to compare with this object.</param>
         /// <returns>
-        /// A 32-bit signed integer that is the hash code for this instance.
-        /// </returns>
+        ///   <see langword="true" /> if the current object is equal to the <paramref name="other" /> parameter; otherwise, <see langword="false" />.</returns>
+        public bool Equals(ResizeLayer other)
+            => other != null
+            && this.Size == other.Size
+            && this.MaxSize == other.MaxSize
+            && this.RestrictedSizes.SequenceEqual(other.RestrictedSizes)
+            && this.ResizeMode == other.ResizeMode
+            && this.AnchorPosition == other.AnchorPosition
+            && this.Upscale == other.Upscale
+            && this.CenterCoordinates == other.CenterCoordinates
+            && this.AnchorPoint == other.AnchorPoint;
+
+        /// <summary>Serves as the default hash function.</summary>
+        /// <returns>A hash code for the current object.</returns>
         public override int GetHashCode()
         {
-            unchecked
-            {
-                int hashCode = this.Size.GetHashCode();
-                hashCode = (hashCode * 397) ^ this.MaxSize.GetHashCode();
-                hashCode = (hashCode * 397) ^ (this.RestrictedSizes?.GetHashCode() ?? 0);
-                hashCode = (hashCode * 397) ^ (int)this.ResizeMode;
-                hashCode = (hashCode * 397) ^ (int)this.AnchorPosition;
-                hashCode = (hashCode * 397) ^ this.Upscale.GetHashCode();
-                hashCode = (hashCode * 397) ^ (this.CenterCoordinates?.GetHashCode() ?? 0);
-                return (hashCode * 397) ^ this.AnchorPoint.GetHashCode();
-            }
+            int hashCode = -1040263854;
+            hashCode = (hashCode * -1521134295) + this.Size.GetHashCode();
+            hashCode = (hashCode * -1521134295) + this.MaxSize.GetHashCode();
+            hashCode = (hashCode * -1521134295) + this.RestrictedSizes.GetHashCode();
```

Comment:
This has the same problem as the `Equals` method and should be updated to this (to my best knowledge of how `GetHashCode` works):
```suggestion
            hashCode = (hashCode * -1521134295) + (this.RestrictedSizes == null ? 0 : this.RestrictedSizes.GetHashCode());
```